drake/solvers/MultipleTreeProblem.m

```diff
@@ -0,0 +1,463 @@
+classdef MultipleTreeProblem
```

Comment:
This is some awesome functionality. I'd love to see it implemented more generally (i.e. supporting any `MotionPlanningTree` class for the sub-trees) eventually. For now, could we rename it to reflect that it's limited to `OptimalTaskSpaceMotionPlanningTree`  sub-trees?